app/Http/Controllers/Sponsors.php

```diff
@@ -152,6 +156,64 @@ private function getSponsors() {
         }
     }
 
+    private function getSponsorDisplay() {
+        // I also add on the SponsorDetail that includes the logo URL + description.
+        if(Auth::check() && in_array(Auth::user()->type, ["admin", "committee","sponsor","hacker"])) {
+            $sponsors = DB::table('sponsors')
+                ->leftJoin('sponsor_details',function($join) {
+                    $desired_type = "portal-info";
+                    $join->on('sponsors.id','=','sponsor_details.sponsor_id')
+                        ->where('sponsor_details.type','=',$desired_type);
+                })
+                ->select('sponsors.id','name','tier','payload','slug')
+                ->get();
+            return response()->json([
+                "success" => true,
+                "data" => $sponsors
+            ]);
+        } else {
+            $this->fail("Unauthorised/unauthenticated.");
+        }
+    }
+
+    private function array_flatten($array) {
+        $return = array();
+        foreach ($array as $key => $value) {
+            if (is_array($value)){ $return = array_merge($return, array_flatten($value));}
+            else {$return[$key] = $value;}
+        }
+        return $return;
+     
+     }
+     
+    private function loadAllResources() {
+        if(Auth::check() && in_array(Auth::user()->type, ["admin", "committee","sponsor","hacker"])) {
+            $sponsors = Sponsor::all();
+            if ($sponsors) {
+                $all_details = array();
```

Comment:
for consistency can we use `$all_details = [];`

I know it is something very minor, but this is what has been in the code so far.